### PR TITLE
docs: add the contribution guideline

### PR DESCRIPTION
**Pull request by Shin-kyoto**
# Description

I want to add `CONTRIBUTING.md`.

# Related links

[TIER IV INTERNAL LINK](https://star4.slack.com/archives/C03AKTRR7AR/p1724028538238149?thread_ts=1724026444.812409&cid=C03AKTRR7AR)

# Notes for reviewers

@mitsudome-r @HansRobo 

- Please check whether the image file's location is appropriate.
- Please check the content of workflow is appropriate.
- Please check the change in GitHub workflow is appropriate.

### CODE REVIEW

File: .github/workflows/add-word.yaml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
           branch: add-word-${{ inputs.word }}
           title: "feat(words): add ${{ inputs.word }}"
           commit-message: "feat(words): add ${{ inputs.word }}"
-          body: "TODO: Update this comment and add some reference links."
+          body: "TODO: Update this comment and add some reference links. \n For reviewer: Please check [CONTRIBUTING.md](https://github.com/tier4/autoware-spell-check-dict/CONTRIBUTING.md)"
```
Comment (HansRobo) [inline .github/workflows/add-word.yaml]: Meybe, the link for CONTRIBUTING.md is invalid.
for example, https://github.com/tier4/autoware-spell-check-dict/README.md is invalid link too.

File: .github/workflows/add-word.yaml [hunk 2]
```diff
@@ -42,7 +42,9 @@ jobs:
           branch: add-word-${{ inputs.word }}
           title: "feat(words): add ${{ inputs.word }}"
           commit-message: "feat(words): add ${{ inputs.word }}"
-          body: "TODO: Update this comment and add some reference links."
+          body: |
+            TODO: Update this comment and add some reference links.
+            Please check [CONTRIBUTING.md](https://github.com/tier4/autoware-spell-check-dict/CONTRIBUTING.md) before sending PR.
```
Comment (HansRobo) [inline .github/workflows/add-word.yaml:47]: Meybe, the link for CONTRIBUTING.md is invalid.
for example, [tier4/autoware-spell-check-dict/README.md](https://github.com/tier4/autoware-spell-check-dict/README.md) is invalid link too.

```suggestion
            Please check [CONTRIBUTING.md](https://github.com/tier4/autoware-spell-check-dict/blob/main/CONTRIBUTING.md) before sending PR.
```

Comment (Shin-kyoto) [reply .github/workflows/add-word.yaml]: Thanks!!
Updated in https://github.com/tier4/autoware-spell-check-dict/pull/763/commits/3c5dd68687723386be9490d279a6d75d0f8aeffc

**Review by HansRobo — APPROVED** [review]
Thanks!!!

**Pull request merged**